lib/reports/BackupReportManager.js

```diff
@@ -0,0 +1,224 @@
+          BackupFailureCount: (resultBackups.length - BackupSuccessCount),
+          FailedBackupList: resultBackups.slice(BackupSuccessCount),
+          threeBackupFailureInRowCount: threeBackupFailureCount
+        };
+        _.assign(instance_record, summary);
+        return instance_record;
+      });
+  }
+  static getInstanceDeleteTime(instance_id) {
+    const criteria = {
+      type: CONST.JOB.SCHEDULED_BACKUP,
+      'data.instance_id': instance_id,
+      'response.delete_backup_status.instance_deleted': true
+    };
+    return Repository.findOne(CONST.DB_MODEL.JOB_RUN_DETAIL, criteria);
+  }
+
+  static getInstanceCreateTime(instance_id) {
+    const criteria = {
+      type: CONST.JOB.SCHEDULED_BACKUP,
+      'data.instance_id': instance_id
+    };
+    return Repository.findOne(CONST.DB_MODEL.JOB, criteria);
+  }
+
+
+  static getInstanceIds(start_time, end_time) {
+    function getInstanceIds(instanceList, offset, modelName, searchCriteria, paginateOpts) {
+      if (offset < 0) {
+        return Promise.resolve();
+      }
+      if (!paginateOpts) {
+        paginateOpts = {
+          records: config.external.mongodb.record_max_fetch_count,
+          offset: offset
+        };
+      } else {
+        _.chain(paginateOpts)
+          .set('offset', offset)
+          .value();
+      }
+      return Repository.search(modelName, searchCriteria, paginateOpts)
+        .then((result) => {
+          instanceList.push.apply(instanceList, _.map(result.list, 'data'));
+          return getInstanceIds(instanceList, result.nextOffset, modelName, searchCriteria, paginateOpts);
+        });
+    }
+    const criteria = {
+      searchBy: {
+        type: CONST.JOB.SCHEDULED_BACKUP
+      },
+      selectBy: {
+        'data.instance_id': 1,
+        'data.service_name': 1,
+        'data.instance_name': 1,
+        'data.plan_id': 1,
+        'data.service_plan_name': 1,
+        'data.space_guid': 1,
+        'data.space_name': 1,
+        'data.organization_guid': 1,
+        'data.organization_name': 1
+      }
+    };
+    if (end_time) {
+      _.assign(criteria.searchBy, {
+        createdAt: {
+          $lt: end_time
+        }
+      });
+    }
+    const paginateOpts = {
+      records: config.external.mongodb.record_max_fetch_count,
+      offset: 0
+    };
+    const result = [];
+    return getInstanceIds(result, 0, CONST.DB_MODEL.JOB, criteria, paginateOpts)
+      .then(() => result);
+  }
+
+  //start time and end_time are Date objects
+  //Aggregated documents of all the triggered backup entries
+  static getBackupTriggerRecord(instance_id, start_time, end_time) {
+    const match = {
+      eventName: 'create_backup',
+      metric: 2,
+      instanceId: instance_id,
+      createdAt: {
+        '$gte': start_time,
+        '$lt': end_time
+      }
+    };
+    const group = {
+      _id: '$response.guid',
+      createdAt: {
+        $first: '$createdAt'
+      }
+    };
+    const sort = {
+      createdAt: 1
```

Comment:
avoid magic numbers - rather use 'asc' or 'desc' which are as well supported and makes it more readable.

Reply:
It seems aggregation sort does not support 'asc' or 'desc'. 
https://docs.mongodb.com/v3.2/reference/operator/aggregation/sort/

Reply:
you can create a constant , with value as 1 and use the constant as it will be easy for the reader of the code to understand the reason of using such value.